drivers/led/issi/is31fl3742a-mono.c

```diff
@@ -82,24 +76,16 @@ void is31fl3742a_select_page(uint8_t addr, uint8_t page) {
 
 void is31fl3742a_write_pwm_buffer(uint8_t addr, uint8_t *pwm_buffer) {
     // Assumes page 0 is already selected.
-    // If any of the transactions fails function returns false.
-    // Transmit PWM registers in 12 transfers of 16 bytes.
-    // i2c_transfer_buffer[] is 20 bytes
-
-    // Iterate over the pwm_buffer contents at 16 byte intervals.
-    for (int i = 0; i < IS31FL3742A_PWM_REGISTER_COUNT; i += 16) {
-        i2c_transfer_buffer[0] = i;
-        // Copy the data from i to i+15.
-        // Device will auto-increment register for data after the first byte
-        // Thus this sets registers 0x00-0x0F, 0x10-0x1F, etc. in one transfer.
-        memcpy(i2c_transfer_buffer + 1, pwm_buffer + i, 16);
+    // Transmit PWM registers in 6 transfers of 30 bytes.
 
+    // Iterate over the pwm_buffer contents at 30 byte intervals.
+    for (uint8_t i = 0; i < IS31FL3742A_PWM_REGISTER_COUNT; i += 30) {
 #if IS31FL3742A_I2C_PERSISTENCE > 0
-        for (uint8_t i = 0; i < IS31FL3742A_I2C_PERSISTENCE; i++) {
-            if (i2c_transmit(addr << 1, i2c_transfer_buffer, 17, IS31FL3742A_I2C_TIMEOUT) != 0) break;
+        for (uint8_t j = 0; j < IS31FL3742A_I2C_PERSISTENCE; j++) {
+            if (i2c_writeReg(addr << 1, i + 1, pwm_buffer + i, 30, IS31FL3742A_I2C_TIMEOUT) != 0) break;
```

Comment:
There is a mixture of `!= 0` and `==0` within the various files, some of which have been updated, assuming this is incorrect and not intentional?